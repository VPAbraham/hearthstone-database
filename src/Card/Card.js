import React from 'react';

const Card = (props) => {
   return (
     <div className="card-display">
       <h4>{props.name}</h4>
       <img src={props.img} alt={props.name}/>
     </div>
   )
}

export default Card;