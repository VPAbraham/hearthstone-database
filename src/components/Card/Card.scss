.card {
  font-family: 'Work Sans', sans-serif;
  width: 18.5rem;
  height: 27rem;
  background: rgb(201, 61, 61);
  border-radius: 8px;
  border: solid 4px rgb(80, 12, 12);
  transition: 0.5s;

  &:hover {
    transition: 0.5s;
    box-shadow: 0px 5px 15px -4px #ebca84;
    border: solid 4px rgb(212, 184, 92);
    width: 19rem;
  }
}