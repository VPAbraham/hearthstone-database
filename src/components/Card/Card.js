import React from 'react';
import './Card.scss';

export const Card = (props) => {
  const { name, imageUrl, pokedexNum,
    hp, attacks, resistances, weaknesses, rarity,
    types, set, artist } = props;

  return(
    <div className='card'>
      <h1>{name}</h1>
      <p>Pokedex No. {pokedexNum}</p>
      <p>Type: {types}</p>
      <p>Rarity: {rarity}</p>
      <img src={imageUrl} alt={name}/>
    </div>
  )
}

export default Card;