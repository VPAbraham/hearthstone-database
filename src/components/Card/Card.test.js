import React from 'react';
import { Card } from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Card 
      name={'Charizard' || 'N/A'}
      imageUrl={'www.image.com' || 'N/A'}
      pokedexNum={123 || 'N/A'}
      hp={'70' || 'N/A'}
      types={['Fire'] || 'N/A'}
      attacks={['Ember', 'Fire Blast'] || 'N/A'}
      resistances={['Rock'] || 'N/A'}
      weaknesses={['Water'] || 'N/A'}
      rarity={'Rare' || 'N/A'}
      key={1}
      set='Base 1'
      artist='Some Guy'
    />)
  })
  it('should match the snapshot with the data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })
})