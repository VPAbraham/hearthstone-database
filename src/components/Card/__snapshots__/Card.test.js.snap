// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with the data passed in 1`] = `
<div
  className="card"
>
  <h1>
    Charizard
  </h1>
  <p>
    Pokedex No. 
    123
  </p>
  <p>
    Type: 
    Fire
  </p>
  <p>
    Rarity: 
    Rare
  </p>
  <img
    alt="Charizard"
    src="www.image.com"
  />
</div>
`;
